test/test_nn.py

```diff
@@ -7699,6 +7701,39 @@ def test_adaptive_log_softmax(self):
         out = asfm.predict(x)
         self.assertEqual(out, asfm.log_prob(x).argmax(dim=1))
 
+    def test_transformer_number_match(self):
+        # Train the simple copy task.
+        V = 11
+        num_encoder_layers = 1
+        num_decoder_layers = 1
+        d_model = 16  # embedding
+        nhead = 2  # number of heads
+        dropout = 0.20
+        d_ff = 64
+        batch_size = 400
+        milliseconds = int(round(time.time() * 1000))
+        torch.manual_seed(milliseconds)
+
+        model = Transformer(V, V, d_model=d_model, nhead=nhead, num_encoder_layers=num_encoder_layers, 
+                            num_decoder_layers=num_decoder_layers, d_ff=d_ff)
+        model_opt = NoamOpt(model.encoder.src_embed.d_model, 1, 400,
+                            torch.optim.Adam(model.parameters(), lr=2.5, betas=(0.9, 0.98), eps=1e-9))
+        criterion = LabelSmoothingLoss(size=V, padding_idx=0, smoothing=0.0)
+
+        print("Training the model...")
+        for epoch in range(10):
+            model.train()
+            train_epoch(data_gen(V, batch_size, 21), model, criterion, model_opt) 
+            model.eval()
+            valid_epoch(data_gen(V, batch_size, 6), model, criterion) 
+
+        model.eval()
+
+        for _ in range(3):    
+            src = Variable(torch.randint(1, V, (10, 1)))
+            src[0][0] = 1
+            tgt = generate_test(model, src, max_len=10, start_symbol=1)
+            assert np.allclose(src, tgt, atol=1e-5) 
```

Comment:
There's a torch "allclose" equivalent. You can't always assume np is available. If you have to rely on numpy, decorate the test to be skipped if numpy isn't available (see other tests for code examples).

Reply:
I will search and make some changes.